src/update_agent/mod.rs

```diff
@@ -117,6 +117,51 @@ impl UpdateAgentState {
         self.transition_to(target);
     }
 
+    /// Record a deploy attempt in UpdateAvailable state.
+    ///
+    /// In case of persistent failure, this returns the update version which
+    /// is being abandoned.
+    fn deploy_attempted(&mut self, success: bool) -> Option<Release> {
+        // Maximum failed deploy attempts before declaring a persistent error.
+        const MAX_DEPLOY_ATTEMPTS: u8 = 60;
```

Comment:
Good catch, I botched this. It was supposed to be 60 minutes. Retrying every 5 minutes, the watermark here should be at `12`.

This cap is a bit arbitrary, anything strictly larger than 1 would indeed work.
A single attempt instead would remove the re-entering edge. With the current ticking logic, it could result in an infinite fast-spinning loop where the agent is oscillating between `NoNewUpdate` and `UpdateaAvailable` as fast as it can.